docs/graph-algorithms/depth-first-search.mdx

```diff
@@ -0,0 +1,346 @@
+![2](https://user-images.githubusercontent.com/70858557/115229939-45b6a800-a131-11eb-9438-4e93dc66cab4.PNG)
+![3](https://user-images.githubusercontent.com/70858557/115229943-46e7d500-a131-11eb-8f3f-d19cf2856d3c.PNG)
+![4](https://user-images.githubusercontent.com/70858557/115229946-47806b80-a131-11eb-8265-924bcc158ddb.PNG)
+![5](https://user-images.githubusercontent.com/70858557/115229947-47806b80-a131-11eb-94a6-fc80ca30fd0a.PNG)
+
+
+## Algorithm
+
+- Set status as ready state for each node in the Graph. 
+- Push the first node to the stack and set its status as waiting state.
+- Pop the top node and after processing set its status as processed state.
+- push on the stack all the neighbours of top node that are in the ready state.
+- change the status of the top node and set its status to waiting state.
+- Repeat the steps until the stack is empty.
+
+
+## Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="c"
+  values={[
+    { label: 'C', value: 'c', },
+    { label: 'C++' , value: 'cpp'}, 
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="c">
+
+```c
+#include<stdio.h>
+#include<stdlib.h>
+#include<unistd.h>
+
+
+#define MAX 20
+#define TRUE 1
+#define FALSE 0
+
+
+int g[MAX][MAX];
+int v[MAX];
+int n;
+
+void main()
+{
+    int v1, v2;
+    char ans;
+    void create();
+    void Dfs(int);
+    create();
+    getchar();
+    do
+    {
+        for (v1 = 0; v1 < n; v1++)
+        v[v1] = FALSE;
+        printf("Enter the Vertex from which you want to traverse ");
+        scanf("%d", &v1);
+        if (v1 >= MAX)
+            printf("Invalid Vertex\n");
+        else
+        {
+            printf("The Depth First Search of the Graph is\n");
+            Dfs(v1);
+        }
+        printf("\nDo you want to traverse By any other node?");
+        ans = getchar();
+    } while (ans == 'y');
+}
+void create()
+{
+    int ch, v1, v2, flag;
+    char ans = 'y';
+    getch();
+
+    for (v1 = 0; v1 < n; v1++)
+        for (v2 = 0; v2 < n; v2++)
+            g[v1][v2] = FALSE;
+    printf("\nEnter no. of nodes");
+    scanf("%d", &n);
+    do
+    {
+        scanf("%d%d", &v1, &v2);
+        if (v1 >= n || v2 >= n)
+            printf("Invalid Vertex Value\n");
+        else
+        {
+            g[v1][v2] = TRUE;
+            g[v2][v1] = TRUE;
+        }
+        printf("\n\nAdd more edges??(y/n)");
+        ans = getchar();
+    } while (ans == 'y');
+}
+void Dfs(int v1)
+{
+    int v2;
+    printf("% d",v1);
+        v[v1] = TRUE;
+    for (v2 = 0; v2 < MAX; v2++)
+        if (g[v1][v2] == TRUE && v[v2] == FALSE)
+            Dfs(v2);
+}
+```
+
+</TabItem>
+
+<TabItem value="cpp">
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+void dfs(int source, vector <int> adj[],bool visited[]){
+    //visit the source
+    cout<<source<<" ";
+    //mark visited as true
+    visited[source] = true;
+    //Traverse all its adjacents
```

Comment:
Missing whitespace and please refactor other ones as well.

Reply:
> Missing whitespace and please refactor other ones as well.

I'm sorry I didn't get it, where the whitespaces are missing ?